scylla/tests/integration/consistency.rs

```diff
@@ -0,0 +1,434 @@
+        rx = check_consistencies(consistency, serial_consistency, rx).await;
+
+        // Set on an exec profile
+        let handle = base_for_every_profile
+            .clone()
+            .consistency(consistency)
+            .serial_consistency(serial_consistency)
+            .build()
+            .into_handle();
+        query_consistency_set_on_exec_profile(&session, &query, handle.clone()).await;
+        rx = check_consistencies(consistency, serial_consistency, rx).await;
+
+        execute_consistency_set_on_exec_profile(&session, &prepared, handle.clone()).await;
+        rx = check_consistencies(consistency, serial_consistency, rx).await;
+
+        batch_consistency_set_on_exec_profile(&session, &batch, handle.clone()).await;
+        rx = check_consistencies(consistency, serial_consistency, rx).await;
+
+        // Set on session's default exec profile
+        let session_with_consistencies = session_builder
+            .clone()
+            .default_execution_profile_handle(handle)
+            .build()
+            .await
+            .unwrap();
+        session_with_consistencies
+            .use_keyspace(ks, true)
+            .await
+            .unwrap();
+        rx = after_session_init(rx).await;
+
+        session_with_consistencies
+            .query(QUERY_STR, (1,))
+            .await
+            .unwrap();
+        rx = check_consistencies(consistency, serial_consistency, rx).await;
+
+        session_with_consistencies
+            .execute(&prepared, (1,))
+            .await
+            .unwrap();
+        rx = check_consistencies(consistency, serial_consistency, rx).await;
+
+        session_with_consistencies
+            .batch(&batch, ((1,),))
+            .await
+            .unwrap();
+        rx = check_consistencies(consistency, serial_consistency, rx).await;
+    }
+}
+
+// Checks that the expected consistency and serial_consistency are set
+//  in the CQL request frame.
+#[tokio::test]
+#[ntest::timeout(60000)]
+#[cfg(not(scylla_cloud_tests))]
+async fn consistency_is_correctly_set_in_cql_requests() {
+    let res = test_with_3_node_cluster(
+        ShardAwareness::QueryNode,
+        |proxy_uris, translation_map, mut running_proxy| async move {
+            let request_rule = |tx| {
+                RequestRule(
+                    Condition::or(
+                        Condition::RequestOpcode(RequestOpcode::Execute),
+                        Condition::or(
+                            Condition::RequestOpcode(RequestOpcode::Batch),
+                            Condition::and(
+                                Condition::RequestOpcode(RequestOpcode::Query),
+                                Condition::BodyContainsCaseSensitive(Box::new(
+                                    *b"INTO consistency_tests",
+                                )),
+                            ),
+                        ),
+                    ),
```

Comment:
https://doc.rust-lang.org/std/ops/index.html

> Note that the && and || operators are currently not supported for overloading. Due to their short circuiting nature, they require a different design from traits for other operators like [BitAnd](https://doc.rust-lang.org/std/ops/trait.BitAnd.html). Designs for them are under discussion.

Interesting, I didn't know that. Maybe we can consider using the `&` and `|` operators instead until they figure the logical ones out.

Reply:
I'm in favour of explicit methods in case of combining `Condition`s. Idiomatic Rust doesn't switch to operator overloading, unlike C++.

Reply:
> I'm in favour of explicit methods in case of combining `Condition`s.

If by methods you mean non-static methods here, then I'd argue that they are similarly awkward as calling them without `self`.

> Idiomatic Rust doesn't switch to operator overloading, unlike C++.

I don't think this is true. Operators are usually overloaded for mathematical objects such as bignums or vectors. In our case, you would construct a predicate, a boolean expression - boolean operator are a natural fit (well, we would need to use the bitwise equivalents but it would be close enough). I don't think it is a big abuse of the syntax.

Ultimately, I think we should just allow closure predicates that takes a context object which allows doing all those checks directly in the code (e.g. `|ctx| match ctx.request_opcode() { ... }`). This would lead to much more natural syntax and greater flexibility.

Reply:
This is anyway out of this PR's scope. For now, let's stick to methods.